src/lib/y2users/linux/writer.rb

```diff
@@ -149,6 +152,13 @@ def write
       # @return [Y2User::Config]
       attr_reader :initial_config
 
+      # Writes the useradd configuration to the system
+      #
+      # @param issues [Y2Issues::List]
+      def write_useradd_config(issues)
+        UseraddConfigWriter.new(config, initial_config).write(issues)
```

Comment:
NP: I would prefer to generate a list of issues (like other writers)  instead of passing the issues list.

Reply:
I did it that way because this writer is called from the more general Linux writer and so far we have followed the approach of creating an empty list at the beginning of the whole process and then passing that list down for each step to accumulate issues in that original list. It's simply not so obvious now because all the steps of the reader are (wrongly, IMHO) implemented via methods, all of them mixed into the same big class, instead of separate smaller classes (like this one).

So the change you suggest would also be inconsistent from some point of view.

I would agree if the current implementation of `Linux::Writer#read` would be something like this:

```ruby
issues = add_groups
issues += write_useradd_config # This instantiates a UseraddConfigWriter internally
issues += add_users
issues += edit_users

issues
```

But there is no method `#+` or `#merge` or similar for `Y2Issues::List`. We could, of course, go all the way down and implement such method and then stop passing the list to the new partial writer AND also to all the methods like `add_groups`. But fixing just the `write_useradd_config` step because is the only one that uses a separate class looks like barking to the wrong tree to me.

Reply:
Yes, it has been pointed several times that it would be nice to split this writer into smaller classes. And I don't get why we decided to pass the *issues* parameter all over the place instead of using an object attribute, something like:

~~~
def write
  @issues = Y2Issues::List.new
  
  add_groups
  write_useradd_config
  ...

  @issues
end
~~~

But in general I agree that ideally we should have something like `Y2Issues::List#concat`. Anyway, IMHO, passing the issues parameter to the new writer is a movement in the opposite way we would like to go. But it is only a detail, nothing super important from my point of view ;)

Reply:
> And I don't get why we decided to pass the _issues_ parameter all over the place instead of using an object attribute,

I did it at the very beginning (to use an object attribute), but it was suggested to use a param instead (for the sake of simplicity, I guess)

Reply:
Since this is extracted to a new card (which would imply the creation of `Y2Issues::List#concat`), I'm leaving it as it is for the time being.

https://trello.com/c/yVrdlyCx/4747-refactor-linuxwriter